### ISSUE TITLE
Inconsistent sys.path setup between Anaconda venvs

### ISSUE THREAD

**Issue by lpuglia** [microsoft/vscode-jupyter#9642]
## Environment data

-   VS Code version: 1.66.1
-   Jupyter Extension version (available under the Extensions sidebar): v2022.3.1000901801
-   Python Extension version (available under the Extensions sidebar): v2022.4.1
-   OS and version: Windows 10
-   Python and/or Anaconda version: conda 4.12.0
-   Type of virtual environment used: conda
-   Jupyter server running:  Remote

## Introduction

when i press on the kernel button above a jupyter notebook i'm presented with the following drop down menu:

![image](https://user-images.githubusercontent.com/2184440/162614446-15edcd92-d437-4911-91bd-9f3f3f0f3b66.png)

the menu is divided in 4 sections, some of the items looks to be repeated but after some trial and error i figure out they have different purposes. If i understand correctly so far i got that:

* section ***(Remote) Jupyter Kernel*** opens a new session that is shown in the ***(Remote) Jupyter Session*** section and doesn't close after close/open of VSCode;
* section ***(Remote) Jupyter Session*** allows you to hijack any open session that is already present on the server. You can distinguish between a session that was opened by Jupyter Notebook App by the relative path in the name of the file:
![image](https://user-images.githubusercontent.com/2184440/162614507-4237e7da-c84e-4c9d-81ee-60e973966fe5.png)
the first one was opened in VSCode Jupyter extension while the second one was open by the official app;
* section ***Conda Env*** shows what conda environments are installed on the server. Opening one of these will result in a non permanent session (not shown in ***(Remote) Jupyter Session***) that can't be recovered after close/open of VSCode;
* section ***Global Env*** pretty much the same as the previous section.

## Expected behaviour

After choosing a kernel from the dropdown menu i expect `sys.path` to be consistent between a kernel created with a ***Conda Env*** menu option and a kernel created with ***(Remote) Jupyter Kernel***

## Actual behaviour

The two `sys.path` are different:
* Scenario 1: A kernel created with ***Conda Env*** menu option contains the path of the current ipynb file;
* Scenario 2: A kernel created with ***(Remote) Jupyter Kernel*** menu option does not contain the said path.

The problem is that in the second scenario, unless i specifically add the current path in `sys.path`, i cannot do relative `import`. I could work with the first scenario, but the problem, as I already explained, is that the kernel is non-permanent and i don't want to loose it between one work session and the other.

I'm not sure how to solve this problem, i saw a lot of references to the option `Jupyter: Notebook File Root` that can be set to either `${workspaceFolder}` or `${fileDirName}`, and i also know that this extension follow VSCode policy of having just the workspace folder as a relative point for the import. But so far i haven't found a solution to my problem. 

What i would like to have is the possibility to create a permanent kernel that contains by default the path of the notebook from which it was openened from. Any help is really appreciated.

My current workaround is to open a kernel from the jupyter app and then connect to it from the VSCode, but that is less than ideal.

**Comment by DonJayamanne** [comment]
thanks for filling this issue.

with kernels from remote paths, there's no way to set the path to be the same as your conda env.

its possible your remote kernel is running on windows, now your path in that kernel is using windows paths in that remote machine.
as for conda environment, it's local and let's assume you have a mac for your local machine. now the paths in your local conda env point to your local mac hence use unix paths. as you can see it's impossible to have local unix paths in your remote windows machine, they are two completely different environments.

basically its not possible to have the same paths.

**Comment by DonJayamanne** [comment]
> that the kernel is non-permanent and i don't want to loose it between one work session and the other.

unfortunately the only way to get around this is to start jupyter locally on your own machine and connect to that jupyter server from within vscode. at that point it works be tested as a remote connection.
however you'd need to setup the paths you're self, as we have no idea where is a local or remote server.

**Comment by DonJayamanne** [comment]
possible duplicate of https://github.com/microsoft/vscode-jupyter/issues/8771
